it/v1/endpoints/RetrievePeriodicObligationsControllerISpec.scala

```diff
@@ -0,0 +1,191 @@
+      "from" -> "2019-01-01",
+      "to" -> "2019-06-06"
+    )
+
+    def request(): WSRequest = {
+      setupStubs()
+      buildRequest(uri)
+        .withHttpHeaders((ACCEPT, "application/vnd.hmrc.1.0+json"))
+    }
+
+    val responseBody = Json.parse(
+      """{
+        |  "obligations": [
+        |     {
+        |       "typeOfBusiness": "self-employment",
+        |       "businessId": "XAIS123456789012",
+        |       "obligationDetails": [
+        |         {
+        |           "periodStartDate": "2019-01-01",
+        |           "periodEndDate": "2019-06-06",
+        |           "dueDate": "2019-04-30",
+        |           "receivedDate": "2019-04-25",
+        |           "status": "Open"
+        |         }
+        |       ]
+        |    }
+        |  ]
+        |}
+        |""".stripMargin)
+
+    val desResponse: JsValue = Json.parse(
+      """
+        |{
+        |    "obligations": [
+        |        {
+        |            "identification": {
+        |                "incomeSourceType": "ITSB",
+        |                "referenceNumber": "XAIS123456789012",
+        |                "referenceType": "IncomeSourceId"
+        |            },
+        |            "obligationDetails": [
+        |                {
+        |                    "status": "O",
+        |                    "inboundCorrespondenceFromDate": "2019-01-01",
+        |                    "inboundCorrespondenceToDate": "2019-06-06",
+        |                    "inboundCorrespondenceDateReceived": "2019-04-25",
+        |                    "inboundCorrespondenceDueDate": "2019-04-30",
+        |                    "periodKey": "#001"
+        |                }
+        |            ]
+        |        }
+        |    ]
+        |}
+        |""".stripMargin
+    )
+
+    def errorBody(code: String): String =
+      s"""
+         |{
+         |     "code": "$code",
+         |     "reason": "des message"
+         |}
+    """.stripMargin
+
+  }
+
+  "Calling the retrieve periodic obligations endpoint" should {
+
+    "return a 200 status code" when {
+
+      "any valid request is made" in new Test {
+
+        override def setupStubs(): StubMapping = {
+          AuditStub.audit()
+          AuthStub.authorised()
+          MtdIdLookupStub.ninoFound(nino)
+          DesStub.onSuccess(DesStub.GET, desUri, queryParams, Status.OK, desResponse)
+        }
+
+        val response: WSResponse = await(request().withQueryStringParameters(
+          "typeOfBusiness" -> typeOfBusiness,
+          "incomeSourceId" -> incomeSourceId,
+          "fromDate" -> fromDate,
+          "toDate" -> toDate,
+          "status" -> status).get())
+
+        response.status shouldBe Status.OK
+        response.json shouldBe responseBody
+        response.header("Content-Type") shouldBe Some("application/json")
+      }
+    }
+
+    "return error according to spec" when {
+
+      "validation error" when {
```

Comment:
You currently only check the validation errors - you need a section for DES errors too (everything in the DES -> MTD mapping)

Reply:
Also include a scenario where DES returns a successful response but we filter out everything (for the new NO_OBLIGATIONS_FOUND error)

Reply:
How odd, i had a section for des and no obligations found, must of got lost